### ISSUE TITLE
Filter foreign key field by null

### ISSUE THREAD

**Issue by mick88** [miki725/django-url-filter#11]
I can't seem to find documentation on filtering fields by null value. By logic, Django's `isnull=True` should work, but it doesn't work for me:
`/api/houses/?ownerid__isnull=True` doesn't filter houses with no owner

**Comment by miki725** [comment]
hmm. `ownerid__isnull=True` should work since I overwrite `isnull` to use `BooleanField` - https://github.com/miki725/django-url-filter/blob/master/url_filter/filters.py#L20

can you provide example how you use your filterset?

**Comment by mick88** [comment]
I simply added `ownerid` as one of filterables field in my viewset:
```python
class HouseViewSet(viewsets.ReadOnlyModelViewSet):
    queryset = models.House.objects.all().select_related('ownerid')
    serializer_class = serializers.HouseSerializer
    filter_fields = [
        'price',
        'parking',
        'ownerid',
        'location',
    ]
```
The query should return 1000 objects, but instead returns all 1500. I can even filter houses by owners with 0 or more score `/api/houses/?ownerid__score__gte=0`, effectively returning houses with ownerid not null (500 results).
Other filters, such as `/api/houses/?ownerid__name=mick88` work correctly and return one item.

Settings:
```python
REST_FRAMEWORK = {
    'DEFAULT_FILTER_BACKENDS': (
        'url_filter.integrations.drf.DjangoFilterBackend',
    ),
    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.PageNumberPagination',
    'PAGE_SIZE': API_ITEMS_PER_PAGE,
}
```

**Comment by miki725** [comment]
Thanks for the info. Ill try to take a look when I can.

**Comment by mick88** [comment]
I stepped through the code and I think I found the issue in the library. `get_spec` appears to raise `SkipFilter`, because "isnull" is not in `self.filters` in this line:

```Python
# url_filter/filtersets/base.py:349
if name not in self.filters:  # name = u'isnull'; self = PlayerFilterSet
    raise SkipFilter
```

Further, I found that this issue only manifests itself on foreign key fields. Filtering by other fields with isnull works correctly, i.e. `/api/houses/?price__isnull=True`

**Comment by miki725** [comment]
oh. that makes sense for foreign keys. I have (had) some updates coming up however didnt get a change to push it in yet due to load at work. Ill try to fix it soon or contributions are welcome.

**Comment by mick88** [comment]
I put in a PR there. Not sure if it's the right way to do it, but it's best I could do and seems to work. I haven't tested it with my project yet, but I added a test which tests my case and it passes.

**Comment by mick88** [comment]
I also discovered that it's impossible to filter by `isnull=False` (unless using workaround `isnull!=true`), because isnull boolean field was required. I put in a fix for this in my PR.

I tested this now with my project and works as intended. Please review my changes and merge.

**Comment by miki725** [comment]
Thanks for the reports and the PR.

I actually had `isnull=False` fixed (https://github.com/miki725/django-url-filter/pull/9/files#diff-1c4aee14792f7a8c40bc2256837aa2c5R22) a while ago however never finished writing tests and so never finished https://github.com/miki725/django-url-filter/pull/9. My plan is to catch up on my open-source project these holidays so hopefully will push a new release with some fixes and new features.

**Comment by miki725** [comment]
@mick88 I solved your issue in https://github.com/miki725/django-url-filter/commit/6330135c33edd15b86ecf481a44444cc96a964cb as part of #9 .

now all of these work on the test project:

```
/one-to-one/restaurants/?place__isnull=True
/one-to-one/restaurants/?place_id__isnull=True
```

I also added tests for your explicit scenario of using FKs integer field directly (in your case `ownerid`). They were added in https://github.com/miki725/django-url-filter/commit/da65e93de7a2152357def83e2afe57a9e55e9074

The only issue now is that `ModelFilterSet` does not add these automatically. In the test project project example of Places and Restaurants only `Restaurant.place` filter is added however `Restaurant.place_id` will not. For that I added a separate ticket #14

Let me know if this ticket is good for closing now. Thanks
